File: backend/src/main/java/es/udc/paproject/backend/model/services/TeamService.java
package es.udc.paproject.backend.model.services;

import java.util.List;

import es.udc.paproject.backend.model.entities.Season;
import es.udc.paproject.backend.model.entities.Team;
import es.udc.paproject.backend.model.entities.User;
import es.udc.paproject.backend.model.exceptions.InstanceNotFoundException;

public interface TeamService {

    void addTeam(Team team);

    void addTeamToSeason(Season season, Team team, User user);

    Team findTeamById(Long teamId) throws InstanceNotFoundException;

    Team findTeamByName(String teamName) throws InstanceNotFoundException;

    List<Team> findAllTeams();
    
    List<Season> findSeasonsToTeam(Long teamId) throws InstanceNotFoundException;
    
    void removeTeam(Long teamId) throws InstanceNotFoundException;

    Team updateTeam(Long teamId, String name) throws InstanceNotFoundException;
}

File: frontend/src/backend/teamService.js
import {config, appFetch} from './appFetch';

export const findAllTeams = (onSuccess) =>
    appFetch('/teams', config('GET'), onSuccess);

export const findTeamById = (id, onSuccess) =>
    appFetch(`/teams/team/${id}`, config('GET'), onSuccess);

export const findTeamByName = (name, onSuccess) =>{
    appFetch(`/teams/name/${name}`, config('GET'), onSuccess);
}

export const addTeam = (team, onSuccess, onErrors) =>
    appFetch('/teams/addTeam/', config('POST', team), onSuccess, onErrors);

export const updateTeam = (team, onSuccess, onErrors) =>
    appFetch(`/teams/update/${team.id}`, config('PUT', team), onSuccess, onErrors);

export const removeTeam = (id, onSuccess, onErrors) =>{
    appFetch(`/teams/remove/${id}`, config('DELETE'), onSuccess, onErrors);
}

//addTeamToSeason
//findTeamsToSeason